smartsim/_core/utils/serialize.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import annotations
+
+import json
+import time
+import typing as t
+from pathlib import Path
+
+import smartsim._core._cli.utils as _utils
+
+if t.TYPE_CHECKING:
+    from smartsim import Experiment
+    from smartsim._core.control.manifest import LaunchedManifest as _Manifest
+    from smartsim.database.orchestrator import Orchestrator
+    from smartsim.entity import DBNode, Ensemble, Model
+    from smartsim.settings.base import BatchSettings, RunSettings
+
+_TStepLaunchMetaData = t.Tuple[
+    t.Optional[str], t.Optional[str], t.Optional[bool], str, str
+]
+
+
+def save_launch_manifest(
+    experiment: Experiment, manifest: _Manifest[_TStepLaunchMetaData]
+) -> None:
+    manifest_dir = Path(experiment.exp_path) / ".smartsim/manifest"
+    manifest_dir.mkdir(parents=True, exist_ok=True)
+    manifest_file = manifest_dir / "manifest.json"
+
+    # FIXME: Did we decide on if this should be uuid/literal count of the
+    #        runs/time since epoch/etc.? Whatever it is, it probably should not
+    #        be secs since epoch as this could lead to conflicts if many
+    #        non-blocking runs are made in quick succession.
+    run_id = int(time.time())
```

Comment:
I am the cause of this confusion! I'm pretty sure that originally we wanted a time. Correct me if I'm wrong @ankona but I'm pretty sure you needed it for something? And then in my fake manifests I wrote integers, which made them seem more like literal counts.

Reply:
now that we are selected entities on name and run_id and displaying both in the selection dropdowns, i'd prefer these be a literal count or uuid simply because i think that would look better when displayed.

Reply:
Since time doesn't usually go backwards, I chose to use the timestamp as the unique identifier when logging details for a run. The integer conversion was just because a per-experiment folder-naming collision at the second level is extremely unlikely and I prefer to avoid special characters in folder names if a human will peruse them. Also, each run would appear _in-order_ in the output directory.

I've [added](https://github.com/CrayLabs/SmartSim/pull/356/files#diff-59a63a8a45350c2c4a46ef5841c206c22b72d74a7230792e37c8a1499c6109f4R252-R254) a `get_ts` function into the helpers that converts a timestamp float to an integer:

![image](https://github.com/CrayLabs/SmartSim/assets/3595025/30589d88-796e-4c70-b61e-1c08d1a96893)

A count, IMO, is more likely to have collision issues w/outputs. A UUID doesn't have collision issues but is longer than necessary, has special characters, and forces a user browsing the folders to hop around to find things in order.

Reply:
Discussed offline and decided to use a uuid for `"run_id"`, but added a `"timestep"` key to the manifest json.